crates/matrix-sdk-sqlite/src/utils.rs

```diff
@@ -136,6 +138,24 @@ pub(crate) trait SqliteAsyncConnExt {
         self.execute_batch("PRAGMA journal_size_limit = 10000000;").await.map_err(Error::from)?;
         Ok(())
     }
+
+    /// Defragment the database and free space on the filesystem.
+    ///
+    /// Only returns an error in tests, otherwise the error is only logged.
+    async fn vacuum(&self) -> Result<()> {
+        if let Err(error) = self.execute_batch("VACUUM").await {
+            // Since this is an optimisation step, do not propagate the error
+            // but log it.
+            #[cfg(not(test))]
+            warn!("Failed to vacuum database: {error}");
+
+            // We want to know if there is an error with this step during tests.
+            #[cfg(test)]
```

Comment:
Can we enable it if we've compiled in debug mode too? (that is, with `#[cfg(debug_assertions)]` too)

Reply:
Done